tests/receivers/jmxreceiver/testdata/resource_metrics/all.yaml

```diff
@@ -0,0 +1,207 @@
+            unit: by
+            type: IntGauge
+          - name: jvm.memory.pool.init
+            description: current memory pool usage
+            unit: by
+            labels:
+              name: CMS Old Gen
+            type: IntGauge
+          - name: jvm.memory.pool.init
+            description: current memory pool usage
+            unit: by
+            labels:
+              name: Par Eden Space
+            type: IntGauge
+          - name: jvm.memory.pool.init
+            description: current memory pool usage
+            unit: by
+            labels:
+              name: Par Survivor Space
+            type: IntGauge
+          - name: jvm.memory.pool.init
+            description: current memory pool usage
+            unit: by
+            labels:
+              name: Code Cache
+            type: IntGauge
+          - name: jvm.memory.nonheap.max
+            description: current non-heap usage
+            unit: by
+            type: IntGauge
+          - name: jvm.classes.loaded
+            description: number of loaded classes
+            unit: 1
+            type: IntGauge
+          - name: jvm.threads.count
+            description: number of threads
+            unit: 1
+            type: IntGauge
+          - name: jvm.memory.heap.committed
+            description: current heap usage
+            unit: by
+            type: IntGauge
+          - name: jvm.memory.pool.committed
+            description: current memory pool usage
+            unit: by
+            labels:
+              name: CMS Old Gen
+            type: IntGauge
+          - name: jvm.memory.pool.committed
+            description: current memory pool usage
+            unit: by
+            labels:
+              name: Par Eden Space
+            type: IntGauge
+          - name: jvm.memory.pool.committed
+            description: current memory pool usage
+            unit: by
+            labels:
+              name: Metaspace
+            type: IntGauge
+          - name: jvm.memory.pool.committed
+            description: current memory pool usage
+            unit: by
+            labels:
+              name: Compressed Class Space
+            type: IntGauge
+          - name: jvm.memory.pool.committed
+            description: current memory pool usage
+            unit: by
+            labels:
+              name: Par Survivor Space
+            type: IntGauge
+          - name: jvm.memory.pool.committed
+            description: current memory pool usage
+            unit: by
+            labels:
+              name: Code Cache
+            type: IntGauge
+          - name: jvm.memory.nonheap.used
+            description: current non-heap usage
+            unit: by
+            type: IntGauge
+          - name: jvm.memory.pool.used
+            description: current memory pool usage
+            unit: by
+            labels:
+              name: CMS Old Gen
+            type: IntGauge
```

Comment:
There doesn't seem to be a value reported for this when I test locally so relaxing the type enforcement is necessary to get this to pass. I guess the current otel-java otlp exporter behavior is to send metrics for all instantiated instruments regardless if there's been an observed value.

```suggestion
```

Reply:
Remove the type verification from this metric